packages/substrate-codecs/src/bitSequence.ts

```diff
@@ -0,0 +1,29 @@
+import { Bytes, Decoder, Encoder, createCodec, createDecoder } from "scale-ts"
+import { compactNumber } from "./compact"
+
+export interface BitSequence {
+  bitsLen: number
+  bytes: Uint8Array
+}
+
+const bitSequenceDecoder: Decoder<BitSequence> = createDecoder((data) => {
+  const bitsLen = compactNumber.dec(data)
+  const bytesLen = Math.ceil(bitsLen / 8)
+  const bytes = Bytes(bytesLen).dec(data)
+  return { bytes, bitsLen }
+})
+
+const bitSequenceEncoder: Encoder<BitSequence> = (input) => {
+  if (input.bitsLen > input.bytes.length * 8)
+    throw new Error(
+      `Not enough bits. (bitsLen:${input.bitsLen}, bytesLen:${input.bytes.length})`,
```

Comment:
```suggestion
      `Not enough bytes. (bitsLen:${input.bitsLen}, bytesLen:${input.bytes.length})`,
```

Reply:
Nope. The error is correct... there are not enough `bits` inside the provided `bytes` 🙂

Reply:
wouldnt it be confusing because they would think you're referring to `bitsLens`